core/trino-spi/src/main/java/io/trino/spi/connector/ColumnMetadata.java

```diff
@@ -31,6 +31,8 @@
 public class ColumnMetadata
 {
     private final String name;
+    // If a mapping from columnName (remote) to mappedName exists, it will be here
+    private final Optional<String> mappedName;
```

Comment:
The optional bit is for non-jdbc based connectors, since identifier mappings are only supported in jdbc based connectors.